### PR TITLE
OT NAT64 implementation

### PR DESCRIPTION
**Pull request by MarGasiorek**
This PR implements modules which allow to use NAT64 from OT stack.

### CODE REVIEW

**Review by canisLupus1313 — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]
Thanks

File: modules/openthread/platform/radio.c [hunk 1]
```diff
@@ -508,19 +514,48 @@ static void openthread_handle_received_frame(otInstance *instance,
...
+	NET_DBG("Sending %s packet to ot stack", is_ip6 ? "IPv6" : "IPv4");
 
 	settings.mPriority = OT_MESSAGE_PRIORITY_NORMAL;
 	settings.mLinkSecurityEnabled = true;
-	message = otIp6NewMessage(instance, &settings);
-	if (message == NULL) {
+
+	message = is_ip6 ? otIp6NewMessage(instance, &settings)
+			 : openthread_ip4_new_msg(instance, &settings);
+	if (!message) {
```
Comment (maciejbaczmanski) [inline modules/openthread/platform/radio.c:558]: could you add a `NET_ERR` message here as well?

**Review by maciejbaczmanski — DISMISSED** [review]
lgtm, one suggestion below

File: modules/openthread/platform/radio.c (hunk 1 above)
Comment (MarGasiorek) [reply modules/openthread/platform/radio.c:558]: Added new `NET_ERR` message when memory allocation fails.

**Review by maciejbaczmanski — APPROVED** [review]
Thanks

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**